ydb/core/tx/schemeshard/schemeshard_info_types.cpp

```diff
@@ -174,6 +174,48 @@ bool TSubDomainInfo::CheckDiskSpaceQuotas(IQuotaCounters* counters) {
     return false;
 }
 
+void TSubDomainInfo::CountDiskSpaceQuotas(IQuotaCounters* counters, const TDiskSpaceQuotas& quotas) {
+    if (quotas.HardQuota != 0) {
+        counters->ChangeDiskSpaceHardQuotaBytes(quotas.HardQuota);
+    }
+    if (quotas.SoftQuota != 0) {
+        counters->ChangeDiskSpaceSoftQuotaBytes(quotas.SoftQuota);
+    }
+    for (const auto& [poolKind, poolQuotas] : quotas.StoragePoolsQuotas) {
+        if (poolQuotas.SoftQuota != 0) {
+            counters->ChangeDiskSpaceSoftQuotaBytes(GetUserFacingStorageType(poolKind), poolQuotas.SoftQuota);
+        }
+    }
+}
+
+void TSubDomainInfo::CountDiskSpaceQuotas(IQuotaCounters* counters, const TDiskSpaceQuotas& prev, const TDiskSpaceQuotas& next) {
+    i64 hardDelta = next.HardQuota - prev.HardQuota;
+    if (hardDelta != 0) {
+        counters->ChangeDiskSpaceHardQuotaBytes(hardDelta);
+    }
+    i64 softDelta = next.SoftQuota - prev.SoftQuota;
+    if (softDelta != 0) {
+        counters->ChangeDiskSpaceSoftQuotaBytes(softDelta);
+    }
+    for (const auto& [poolKind, newPoolQuotas] : next.StoragePoolsQuotas) {
+        const auto* oldPoolQuotas = prev.StoragePoolsQuotas.FindPtr(poolKind);
+        i64 delta = newPoolQuotas.SoftQuota - (oldPoolQuotas ? oldPoolQuotas->SoftQuota : 0u);
+        if (delta != 0) {
+            counters->ChangeDiskSpaceSoftQuotaBytes(GetUserFacingStorageType(poolKind), delta);
+        }
+    }
+    for (const auto& [poolKind, oldPoolQuotas] : prev.StoragePoolsQuotas) {
+        if (const auto* newPoolQuotas = next.StoragePoolsQuotas.FindPtr(poolKind);
+            !newPoolQuotas
+        ) {
+            i64 delta = -i64(oldPoolQuotas.SoftQuota);
+            if (delta != 0) {
+                counters->ChangeDiskSpaceSoftQuotaBytes(GetUserFacingStorageType(poolKind), delta);
```

Comment:
This will not get expected result either.
Instead of addition of the negative value, there should be proper subtraction of the positive value.

Reply:
I changed the signature of this function to accept ui64 as a parameter. It should make all the calculations in ui64 and if we assume that all the deltas are correct, then it should never be "negative" (i.e. a very large unsigned number) and should always display correct and expected value.